Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm

```diff
@@ -662,14 +662,30 @@ static void registerLogHook()
     os_log_set_hook(OS_LOG_TYPE_DEFAULT, ^(os_log_type_t type, os_log_message_t msg) {
         if (msg->buffer_sz > 1024)
             return;
-        char* messageString = os_log_copy_message_string(msg);
```

Comment:
Does the hook get called on the main thread? If so, can use ASSERT() that it is the case? Otherwise, it won't be safe to get the NetworkProcess IPC::Connection.

Reply:
Ah, that is a good catch! The hook will be called on the same thread as the logging took place on. I have addressed this issue in the latest patch.